ion/marshal.go

```diff
@@ -308,8 +307,8 @@ func (m *Encoder) encodeStruct(v reflect.Value) error {
 	}
 
 	t := v.Type()
-	if t == timeType {
-		return m.encodeTime(v)
+	if t == timestampType {
```

Comment:
Looks like this would require us to formally add time.TIme as an ion type. 

@therapon, @zslayton What do you think?

Reply:
I'm in favor of also supporting the convenient native types and documenting why those types could be lossy. However, I don't have a strong sense of the level of effort involved. Could we easily add that support in the near future, or does it need to be added now?

> Looks like this would require us to formally add time.TIme as an ion type.

Can you elaborate on this? I read "formally add ... an Ion type" as changing the Ion spec, which is probably not what you're describing.

Reply:
Well implementing this for go time would involve treating time the same way we're treating other ion types such adding a TimeType case to:

![image](https://user-images.githubusercontent.com/62349012/88729587-1f576500-d0e9-11ea-877b-39537156dddc.png)

Reply:
I don't think so? The Ion data will still have TimestampType and get routed into decodeTimestampTo; that's already checking the type of v and we can have cases there for both `v.Type() == timeType` and `v.Type() == timestampType`.

Reply:
Supporting this will require more effort than I want to include in this PR so I created a new issue for it.

https://github.com/amzn/ion-go/issues/113

Reply:
Thanks for opening an issue, I think we can add that in another PR.